mm2src/coins/eth/web3_transport.rs

```diff
@@ -159,13 +221,22 @@ async fn send_request(
     const REQUEST_TIMEOUT_S: f64 = 60.;
 
     let mut errors = Vec::new();
-    for uri in uris.iter() {
-        let request = to_string(&request);
-        event_handlers.on_outgoing_request(request.as_bytes());
 
-        let mut req = http::Request::new(request.clone().into_bytes());
+    let serialized_request = to_string(&request);
+
+    for node in nodes.iter() {
+        let serialized_request =
+            match handle_gui_auth_payload_if_activated(&gui_auth_validation_generator, node, &request, &mut errors) {
+                Ok(r) => r,
+                Err(true) => serialized_request.clone(),
```

Comment:
`handle_gui_auth_payload_if_activated` can return `Ok(to_string(request))` in case `gui_auth` is false. This will avoid this cloning and will lead to removing`let serialized_request = to_string(&request);` avoiding calling `to_string` twice if `gui_auth` is true and there are no errors. Also in this case, `handle_gui_auth_payload_if_activated` error return type doesn't have to be bool.

Reply:
I think using `to_string` for each iteration is a little bit overhead. That's why I moved `to_string` to outside of the for loop and used `clone` instead.

The return type is updated, it's not `bool` anymore. Whole function refactored little bit more.